vars/getFunctionalArgs.groovy

```diff
@@ -0,0 +1,49 @@
+// vars/getFunctionalArgs.groovy
+
+/**
+ * getFunctionalArgs.groovy
+ *
+ * Get a map of arguments used to run the functional tests in the stage.
+ *
+ * @param pragma_suffix commit pragma suffix for this stage, e.g. '-hw-large'
+ * @param stage_tags launch.py tags to use to limit functional tests to those that fit the stage
+ * @param default_tags default launch.py tags to use when no parameter or commit pragma tags exist
+ * @param nvme default launch.py --nvme argument to use
+ * @param provider default launch.py --provider argument to use
+ * @return Map values to use with runTestFunctional
+ */
+Map call(String pragma_suffix, String stage_tags, String default_tags, String nvme,
```

Comment:
Generally we use "Map call(Map config = [:])"  If one of these call signatures does not exist, and you are missing one of your parameters, the Jenkins job will crash with a stack dump claiming this routine does not exist.

Then we do the checks on the incoming config Map members to determine if we have all the needed parameters, etc, and report any problems or conflicts.

This is mainly needed for methods that may be called from Jenkinsfiles, as that is more likely where the programming error may show up.

Reply:
I understand the 'routine does not exist' error - since in grovvy you can define the same function with different inputs, so technically the version w/o all the required inputs does not exist - but using a Map and then having to verify that all the inputs have been specified seems like a lot of extra work just to avoid a failure introduced by a misuse of the function.

Reply:
The updated implementation now uses Map inputs.